userspace/libsinsp/threadinfo.h

```diff
@@ -65,6 +65,21 @@ struct erase_fd_params
 class SINSP_PUBLIC sinsp_threadinfo : public libsinsp::state::table_entry
 {
 public:
+	typedef struct sinsp_userinfo
+	{
+		uint32_t uid; ///< User ID
+		uint32_t gid; ///< Group ID
+		std::string name; ///< Username
```

Comment:
Am not sure whether using string is what we want here; but perhaps it works fine (ie: have you already tested memory consumption with this change?)
Anyway, storing `<NA>` when details are not enabled should not cost that much, right? Returning an empty string breaks UX since we always used `NA` for that.

Reply:
Yes, I've re-run memory test with this change, getting the same results. From what I understand,  and empty`string` doesn't introduce as much overhead as statically allocated array anyway (although it might be larger than some alternatives).

Let me do couple more experiments to estimate how much exactly an overhead from using string and initializing it with NA would be. Another option to this could be to turn this structure into a class with a method `getName`, that will return NA if the underlying string is empty. This would seem to satisfy both memory and UX sides, but will involve more changes. What do you think?

Reply:
> Yes, I've re-run memory test with this change, getting the same results. From what I understand, and emptystring doesn't introduce as much overhead as statically allocated array anyway (although it might be larger than some alternatives).

+1 thanks!

> Another option to this could be to turn this structure into a class with a method getName, that will return NA if the underlying string is empty.

Let's first try if storing `<NA>` in the string causes overhead, otherwise yep this would be an ok solution for me!
Thank you very much for the deep dive btw!